### ISSUE TITLE
Consider using css custom properties for dynamic styles

### ISSUE THREAD

**Issue by TxHawks** [kripod/glaze#24]
## Motivation

Using css custom properties to set custom styles could possibly help minimising the runtime even further. [Linaria](https://github.com/callstack/linaria) is taking this approach, and it would be great to have a combination of their approach with a utility-first approach

**Comment by kripod** [comment]
Thank you for the valuable suggestion! Quoting from the [backstory behind treat](https://seek-oss.github.io/treat/background#backstory):

> [Linaria](https://linaria.now.sh/) looked a lot more promising due to its theming support, but it came with an important caveat. Theming is achieved via CSS custom properties (i.e. CSS variables), which means that legacy browsers like IE11 are not supported. This simply wasn’t an option for us.

To keep glaze working with IE11, custom properties are not being used as of today. However, switching may be an option for the future, e.g. applying rules like:

```jsx
<p className=“pl pr” style={{ '--pl': '1rem', '--pr': '1rem'>…</p>
```

```css
.pl { padding-left: var(--pl); }
.pr { padding-right: var(--pr); }
```

The consequence is that way less CSS classes need to be generated. However, this comes at the cost of a changed API, as a `style` prop shall be specified most of the time.

I think that #9 could solve the concerns regarding bundle size. This should be documented, though.

**Comment by TxHawks** [comment]
Thanks for the detailed response. I understand that IE11 support is crucial for many, but I think it is worth taking into account that many website no longer support IE11. It might be worthwhile making this opt in, probably with a second hook.

I realize this makes the API surface larger and increases code complexity, but I think it could be worth it from future-proofing and performance points of view.

**Comment by kripod** [comment]
I agree, this would be very useful. Currently, I’m working on multi-page docs and then I’ll review #10. Afterwards, the `useStyling()` hook will be re-architectured for improved modularity. I’ll take this use-case into account while redesigning the internals.

**Comment by kripod** [comment]
A very interesting take was posted by @jamiebuilds today: https://twitter.com/buildsghost/status/1251572060750409728

![image](https://user-images.githubusercontent.com/14854048/79671114-23a9e300-81c8-11ea-810f-722858f336d0.png)

**Comment by finnhvman** [comment]
Some thoughts about supporting/not supporting IE11:
https://medium.com/@burger.neal/the-end-of-life-of-internet-explorer-11-12736f9ff75f

**Comment by hennessyevan** [comment]
Could we not put this behind a flag? Or support it with a fallback?

**Comment by kripod** [comment]
Great ideas, thanks! Although I think these should be considered after the project's first stable release. There are a lot of other tasks to do until then, especially regarding the documentation.
